ui/src/criteria/attribute.tsx

```diff
@@ -81,73 +136,216 @@ function AttributeEdit(props: AttributeEditProps) {
+    const selectionIndex = (hint: tanagra.EnumHintValue) =>
+      props.data.selected.findIndex(
+        (row) => row.id === hint.attributeValue?.int64Val
+      );
 
-  if (enumHintValues?.length === 0) {
     return (
-      <Typography>
-        No information for attribute {props.data.attribute}.
-      </Typography>
+      <>
+        {enumHintValues?.map((hint: tanagra.EnumHintValue) => (
+          <ListItem key={hintDisplayName(hint)}>
+            <FormControlLabel
+              label={hintDisplayName(hint)}
+              control={
+                <Checkbox
+                  size="small"
+                  checked={selectionIndex(hint) > -1}
+                  onChange={() => {
+                    props.dispatchFn(
+                      produce(props.data, (data) => {
+                        if (selectionIndex(hint) > -1) {
+                          data.selected.splice(selectionIndex(hint), 1);
+                        } else {
+                          data.selected.push({
+                            id: hint.attributeValue?.int64Val || -1,
+                            name: hintDisplayName(hint),
+                          });
+                        }
+                      })
+                    );
+                  }}
+                />
+              }
+            />
+          </ListItem>
+        ))}
+      </>
     );
   }
+
   return (
-    <>
-      {enumHintValues?.map((hint: tanagra.EnumHintValue) => (
-        <ListItem key={hintDisplayName(hint)}>
-          <FormControlLabel
-            label={hintDisplayName(hint)}
-            control={
-              <Checkbox
-                size="small"
-                checked={selectionIndex(hint) > -1}
-                onChange={() => {
-                  props.dispatchFn(
-                    produce(props.data, (data) => {
-                      if (selectionIndex(hint) > -1) {
-                        data.selected.splice(selectionIndex(hint), 1);
-                      } else {
-                        data.selected.push({
-                          id: hint.attributeValue?.int64Val || -1,
-                          name: hintDisplayName(hint),
-                        });
-                      }
-                    })
-                  );
-                }}
-              />
-            }
-          />
-        </ListItem>
-      ))}
-    </>
+    <Typography>
+      No information for attribute {props.data.attribute}.
+    </Typography>
   );
 }
 
-type ConceptDetailsProps = {
+type AttributeDetailsProps = {
   data: Data;
 };
 
-function ConceptDetails(props: ConceptDetailsProps) {
-  return (
-    <>
-      {props.data.selected.length === 0 ? (
-        <Typography variant="body1">None selected</Typography>
-      ) : (
-        props.data.selected.map(({ id, name }) => (
+function AttributeDetails(props: AttributeDetailsProps) {
+  if (
+    typeof props.data.selected !== "undefined" &&
+    props.data.selected.length !== 0
```

Comment:
`if (props.data.selected?.length)` will cover both cases here and I think satisfy the compiler.